### PR TITLE
build: upgrade Modin and pandas versions

### LINKED ISSUE #1001: Latest Modin as a data source for pandas-ai
### 🚀 The feature

Hey folks, I just saw you had added support for Modin as a data source for pandas-ai in https://github.com/Sinaptik-AI/pandas-ai/pull/907. This is a great addition to the other data sources like pandas and polars, for instance, that has already been introduced much earlier. I am a little concerned about the Modin version pandas-ai currently supports, namely, [0.18.1](https://github.com/Sinaptik-AI/pandas-ai/blob/66dea557956a6e378856472013340c2ced9215c4/poetry.lock#L3369). It is too old and does not contain latest perfomance features and optimizations. Would it be possible to support a latest Modin version?

### Motivation, pitch

All libraries are advancing and making great progress with respect to ultimate performance on latest versions and pandas with Modin are not exceptions. To speed up chatting with data through pandas-ai and reduce users' waiting time it would be great to support latest versions of Modin, pandas, etc.

### Alternatives

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by YarShev**
- [x] Closes #1001
- [ ] Tests added and passed if fixing a bug or adding a new feature.
- [ ] All [code checks passed](https://github.com/gventuri/pandas-ai/blob/main/CONTRIBUTING.md#-testing).

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -39,7 +39,7 @@ streamlit = {version = "^1.23.1", optional = true}
 text-generation = { version = ">=0.6.0", optional = true }
 openpyxl = { version = "^3.0.7", optional = true }
 pymysql = { version = "^1.1.0", optional = true }
-psycopg2 = { version = "^2.9.7", optional = true }
+psycopg2-binary = { version = "^2.9.7", optional = true }
```
Comment (YarShev) [inline pyproject.toml:42]: See the details in https://github.com/Sinaptik-AI/pandas-ai/issues/1032 on why this is changed.

**Comment by gventuri** [comment]
@YarShev thanks a lot for the PR, merging!!

**Comment by YarShev** [comment]
@gventuri, do you think what release of pandas-ai will contain this support for the new versions of Modin and pandas?

**Comment by gventuri** [comment]
2.0.21, it's already out :D

**Comment by YarShev** [comment]
@gventuri, oh, I see :D Thanks a lot ;)

**Pull request merged**